src/content/docs/guides/Beginners/Your First Shader/3_composite_lighting.mdx

```diff
@@ -0,0 +1,134 @@
+---
+title: Lighting in Composites
+description: Implement basic diffuse shading in a composite pass
+sidebar:
+  label: Lighting in Composites
+  order: 3
+---
+
+In this section we will implement basic diffuse shading in the composite pass.
+
+## Gamma Correction
+Before we start doing any lighting, it's important to make sure we are in the correct color space. For more information on why we need to do this, see the relevant [LearnOpenGL article](https://learnopengl.com/Advanced-Lighting/Gamma-Correction).
+
+Anyway, let's go back to our `gbuffers_terrain.fsh`. In order to apply gamma correction, we raise our colour to the power of 2.2. So, at the end of the pass:
+
+```glsl
+color.rgb = pow(color.rgb, vec3(2.2));
+```
+
+You'll notice this makes our colours look horrible. This is because we need to apply inverse gamma correction afterwards. Let's make a new shader pass called `final`. Create copies of your `composite.fsh` and `composite.vsh` files and rename them to `final.fsh` and `final.vsh`. The `final` pass is functionally identical to the `composite` passes but it always runs after the last composite pass.
+
+Then, in your new `final.fsh`:
+```glsl
+color.rgb = pow(color.rgb, vec3(1.0/2.2));
+```
+
+Voila! Your colours are back to normal.
+
+## Decoding Data
+Let's read back in the lightmap and normal data we stored in `colortex1` and `colortex2` in the previous tutorial. First, we need to be able to sample the textures.
+```glsl
+uniform sampler2D colortex1;
+uniform sampler2D colortex2;
+```
+
+Then, in `main`, let's read them back:
+
+```glsl
+vec2 lightmap = texture(colortex1, texcoord).rg; // we only need the r and g components
```

Comment:
Won't this error? For me it does

Reply:
It says I can't convert the vec4 to vec2

Reply:
Did you miss the `.rg`?